warehouse/static/js/warehouse/controllers/password_match_controller.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PasswordController from "./password_controller";
+
+export default class extends PasswordController {
+  static targets = PasswordController.targets.concat(["newPassword", "confirmPassword", "passwordsMatchMessage", "submit"]);
+
+  connect() {
+    // Reset these so they don't persist between page reloads
+    // We assume that both targets above exist
+    super.connect();
+    this.checkPasswordsMatch();
+  }
+
+  checkPasswordsMatch() {
+    if (this.passwordTargets.every(field => field.value === "")) {
+      this.passwordsMatchMessageTarget.classList.add("hidden");
+      this.submitTarget.setAttribute("disabled", "");
+    } else {
+      this.passwordsMatchMessageTarget.classList.remove("hidden");
+      if (this.newPasswordTarget.value === this.confirmPasswordTarget.value) {
```

Comment:
Instead of just checking these two targets, we probably want to simplify this to make sure all `passwordMatchTargets` have the same non-empty value. That way we can reuse this on other pages just by applying the target to whichever password fields we want to match.